plaso/engine/artifacts_trie.py

```diff
@@ -0,0 +1,304 @@
+      # sanitizes path segments before writing data to disk.
+      sanitized_child_segment = path_helper.PathHelper.SanitizePathSegments(
+              [child_segment]).pop()
+      if segment in (child_segment, sanitized_child_segment):
+        # If the child is an exact match, continue traversal.
+        self._SearchTrie(child_node, self._CustomPathJoin(
+            path_separator,
+            current_path, child_segment), remaining_segments, path_separator,
+            matching_artifacts)
+      # If the child is a glob, see if it matches.
+      elif glob.has_magic(child_segment):
+        if self._MatchesGlobPattern(
+                child_segment, segment, child_node.path_separator):
+          self._SearchTrie(child_node, self._CustomPathJoin(
+              path_separator,
+              current_path, segment), remaining_segments, path_separator,
+              matching_artifacts)
+          self._SearchTrie(
+              node,
+              self._CustomPathJoin(
+                  path_separator,
+                  current_path, segment), remaining_segments, path_separator,
+              matching_artifacts)
+
+  def _ComparePathIfSanitized(
+      self, current_path, path_separator, artifact_path,
+      artifact_path_seperator):
+    """Compares a current path with an artifact path, handling sanitization.
+
+    This method checks if the current_path matches the artifact_path,
+    considering that the artifact_path might have been sanitized.
+
+    Args:
+      current_path (str): The current path being checked.
+      path_separator (str): Path separator for the current path.
+      artifact_path (str): The artifact path to compare against.
+      artifact_path_seperator (str): Path separator for the artifact path.
+
+    Returns:
+      bool: True if the current path matches the artifact path or its
+          sanitized version, False otherwise.
+    """
+    artifact_path_segments = self._GetNonEmptyPathSegments(
+        artifact_path, artifact_path_seperator)
+    sanitized_path_segments = path_helper.PathHelper.SanitizePathSegments(
+        artifact_path_segments)
+    return self._GetNonEmptyPathSegments(current_path, path_separator) in [
+        artifact_path_segments, sanitized_path_segments]
+
+  def _GetNonEmptyPathSegments(self, path, separator):
+    """Splits a path into segments and remove non-empty segments.
+
+    Args:
+        path (str): The path string to be split.
+        separator (str): The path separator.
+
+    Returns:
+        list[str]: A list of non-empty path segments.
+    """
+    return [s for s in path.split(separator) if s]
+
+  def _GetArtifactsPaths(self, node):
+    """Retrieves a mapping of artifact names to their paths.
+
+    Args:
+      node (TrieNode): current trie node being traversed.
+
+    Returns:
+        dict: dictionary mapping artifact names to their paths.
+    """
+    artifacts_paths = {}
+
+    def _collect_paths(node, current_path, artifacts):
```

Comment:
could this be moved into a class method ?

Reply:
done